vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,137 @@
+package vsphere
+
+import (
+	"fmt"
+
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+)
+
+func resourceVSphereHostVirtualSwitch() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the virtual switch.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host_system_id": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The managed object ID of the host to set the virtual switch up on.",
+			Required:    true,
+			ForceNew:    true,
+		},
+	}
+	mergeSchema(s, schemaHostVirtualSwitchSpec())
+
+	// Transform any necessary fields in the schema that need to be updated
+	// specifically for this resource.
+	s["active_nics"].Required = true
+	s["standby_nics"].Required = true
+
+	s["teaming_policy"].Default = hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcID
+	s["check_beacon"].Default = false
+	s["notify_switches"].Default = true
+	s["failback"].Default = true
+
+	s["allow_promiscuous"].Default = false
+	s["allow_forged_transmits"].Default = true
+	s["allow_mac_changes"].Default = true
+
+	s["shaping_enabled"].Default = false
```

Comment:
Yeah, this is the best way I could think of doing it for now as these items are shared between resources, and the defaults can't be defined in the other resource so that `GetOkExists` works correctly (or at least my understanding of `GetOkExists`).

It would be a neat idea if eventually we got to a place where we could _transform_ schemas where we could have a base schema, and then make changes to items in the base via something a little more substantial than a single assignment.

Reply:
yeah that sounds sensible. Agree about transforming schema's - there's a couple of other options too given the number of fields this could work with a model object or even just using constants for the field names used throughout the codebase (I've been bitten by magic strings too many times!) - but that's not a blocker by any means just one to 🤔 about :)